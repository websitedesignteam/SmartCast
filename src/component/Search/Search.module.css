.overlayContent{
       width: 100%;
       text-align: center;
       margin-top: 50px; 
}

.searchBar{
       width: 90%;
       height: 35px;
}

.searchButton{
       height: 41px;
}

.checkboxes{
       display: flex;
       flex-direction: row;
       margin-left: 35px;
       margin-top: 15px;
}


