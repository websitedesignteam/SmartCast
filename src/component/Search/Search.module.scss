@import "../../styles/variables";

input:-webkit-autofill,
input:-webkit-autofill:hover,
input:-webkit-autofill:focus textarea:-webkit-autofill,
textarea:-webkit-autofill:hover textarea:-webkit-autofill:focus,
select:-webkit-autofill,
select:-webkit-autofill:hover,
select:-webkit-autofill:focus {
	-webkit-box-shadow: 0 0 0px 1000px $gray inset !important;
}

.overlayContent{
       margin-bottom:10px;
}

.checkboxes {
	margin-left: 1.5rem;
}

.searchBar{
	margin: 25px 25px 0px 25px;
	position : relative;
	height: 35px;
	border: 3px solid rgb(139, 188, 226);
	border-radius: 1.5rem;
	-webkit-box-shadow: inset 0px 0px 16px -14px #000000; 
	box-shadow: inset 0px 0px 16px -14px #000000;
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: flex-start;
}

.searchBar:hover, .searchBar:focus {
	box-shadow:0 1px 6px rgba(32,33,36,.28)
}

.searchIcon {
	height: 1rem;
	width: 1rem;
	margin: 0 0.5rem 0 1rem;
	// float: left;
}

.searchInput {
	background: transparent;
	background-color: transparent;
	border: none;
	height: 95%;
	width: 100%;
	margin-right: 1.5rem;
}

input:-webkit-autofill,
input:-webkit-autofill:hover,
input:-webkit-autofill:focus,
input:-webkit-autofill:active {
    -webkit-transition: "color 9999s ease-out, background-color 9999s ease-out";
    -webkit-transition-delay: 9999s;
}

.searchInput::placeholder,
.searchInput::-webkit-input-placeholder,
.searchInput::-moz-placeholder,
.searchInput:-ms-input-placeholder,
.searchInput:-moz-placeholder {
	font-family: "Cairo";
	// color: $dark-gray;
}

.searchInput:focus {
	outline: none;
}

.searchEnter {
	display: none;
}

.checkboxes{
	display: flex;
	flex-direction: row;
	margin-top: 0;
	margin-bottom: 1rem;
	label {
		margin-right: 0.5rem;
		white-space: nowrap;
	}
}


