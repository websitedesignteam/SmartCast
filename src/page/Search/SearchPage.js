import React, {useEffect, useState} from 'react'
import { Link, useParams } from "react-router-dom"
import { searchEpisodes, searchPodcasts, searchTags } from '../../utils/api'
import PodcastContainer from "component/PodcastContainer/PodcastContainer"
import EpisodeContainer from "component/EpisodeContainer/EpisodeContainer"
import styles from './SearchPage.module.scss'
import {baseUrl, errorSearch} from "../../utils/constants";

function SearchPage(props) {
	const { searchTerm, searchType } = useParams();

	const [episodes, setEpisodes] = useState([])
	const [podcasts, setPodcasts] = useState([])
	const [isLoading, setIsLoading] = useState(false)

	useEffect(() => {
		setIsLoading(true);
		let body = {"searchString": searchTerm}
		if (searchType === 'episodes'){
			searchEpisodes(body, 1) 
			.then((response) => {
				setIsLoading(false);
				setEpisodes(response.data.Data.episodes)
			})
			.catch((error) => {
				console.log(error);
			});
		} else if (searchType === 'podcasts'){
			searchPodcasts(body, 1) 
			.then((response) => {
				setIsLoading(false);
				setPodcasts(response.data.Data.podcasts)
			})
			.catch((error) => {
				setIsLoading(false);
				console.log(error);
			});
		} else if (searchType === 'tags'){
			searchTags(body) 
			.then((response) => {
				setIsLoading(false);
				setEpisodes(response.data.Data)
			})
			.catch((error) => {
				setIsLoading(false);
				console.log(error);
			});
		}
	}, [searchTerm, searchType])

	if (isLoading === false){
		if ((searchType === 'episodes' || searchType === 'tags') && episodes.length > 1){
			return( <div>
				<h4>{searchType === 'tags' && "Transcribed"}
					Episode Results for: "{searchTerm}"</h4>
				<div className={styles.cardContainer}>{episodes.map((episode)=>{
					return (
						(searchType === 'tags') 
						? <Link to={{
							pathname: `/podcast/${episode.podcastID}/episode/${episode.episodeID}`,
							episode,
							}} 
							className={styles.link}
						>
							<EpisodeContainer 
								episodeTitle={episode.episodeTitle}
								episodeDescription={episode.episodeDescription}
								imgSrc={episode.episodeImage}
							/>
						</Link> 
						: <Link to={`/podcast/${episode.podcastID}/episode/${episode.episodeID}`} className={styles.link}>
							<EpisodeContainer 
								episodeTitle={episode.episodeTitle}
								episodeDescription={episode.episodeDescription}
								imgSrc={episode.episodeImage}
							/>
						</Link> 
					)
				})}</div>
				</div>)
		} else if (searchType === 'podcasts' && podcasts.length > 1){
			return( <div>
				<h4>Podcast Results for: "{searchTerm}"</h4>
				<div>{podcasts.map((podcast)=>{
					return (
						<Link to={`/podcast/${podcast.podcastID}`} className={styles.link}>
							<PodcastContainer 
								podcastTitle={podcast.podcastTitle}
								podcastDescription={podcast.podcastDescription}
								imgSrc={podcast.podcastImage}
								podcastTotalEpisodes={podcast.totalEpisodes}
							/>
						</Link>
					)
				})}</div>
			</div>)

		} else {
			return (
			<div className={styles.error}>
				<div className={styles.errorText}>
					{errorSearch}
				</div>
				<img className={styles.errorImg} src={baseUrl+'/assets/empty_search.gif'} alt=""/>
			</div>
		);
		}
	} else {
		return (<div className="loader" />)
	}
}

export default SearchPage;
