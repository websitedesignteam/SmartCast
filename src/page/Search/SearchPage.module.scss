@import "../../styles/variables";

.link {
    text-decoration: none;
    color: black;
}

.error {
    display: flex;
    flex-direction: column;
    align-content: center;
}

.errorText {
    margin-top: 1rem;
    margin-bottom: 1rem;
    padding: 1rem;
    border-radius: 1rem;
    background-color: $light-gray;
    font-weight: 700;
    
}

.errorImg {
    width: 100%;
    max-width: 30rem;
    margin: auto;
}

.pageButtons {
	display: flex;
	flex-direction: row;
	width: 100%;
	align-items: center;
	justify-content: center;
	button {
		border-radius: 100%;
		height: 2.5rem;
		width: 2.5rem;
		margin-right: 1rem;
	}
	img {
		height: 1.5rem;
		width: 1.5rem;
	}
	img.lastPage {
		margin: 0 0 -0.4rem -0.2rem;
	}
	img.nextPage {
		margin: 0 -0.2rem -0.4rem 0;
	}
}

// @for $i from 1 through 10 {
//     .cardContainer:nth-child(#{$i})::after{
//         animation-delay: $i *1s;
//     }
// }

// .episodeContainer{
//     display: flex;
// 	flex-direction: row;
// 	border: 0px solid black;
// 	border-radius: 15px;
// 	margin-bottom: 1.5rem;
// 	height: 200px;
// 	padding: 1.5rem;
// 	-webkit-box-shadow: $box-shadow;
// 	-moz-box-shadow: $box-shadow;
// 	box-shadow: $box-shadow;
// 	animation-name: slideRight;
// 	-webkit-animation-name: slideRight;	

// 	animation-duration: 1.3s;	
// 	-webkit-animation-duration: 1.3s;

// 	animation-timing-function: ease-in;
// 	-webkit-animation-timing-function: ease-in;		

// 	visibility: visible !important;	
// }

// @keyframes slideRight {
// 	0% {
// 		transform: translateX(-90%);
// 	}		
// 	100% {
// 		transform: translateX(0%);
// 	}		
// }

// @-webkit-keyframes slideRight {
// 	0% {
// 		-webkit-transform: translateX(-90%);
// 	}	
// 	100% {
// 		-webkit-transform: translateX(0%);
// 	}
// }