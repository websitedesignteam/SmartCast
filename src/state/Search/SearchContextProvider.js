import React, {createContext, useState} from 'react';

export const SearchContext = createContext(null)

function SearchContextProvider(props) {
	const [disableCheckbox, setDisableCheckbox] = useState(true);
	const [searchInput, setSearchInput] = useState("");
	const [searchType, setSearchType] = useState("tags");
	const userContext = {
		searchInput,
		setSearchInput,
		searchType,
		setSearchType,
		disableCheckbox,
		setDisableCheckbox,
	}

	return (
		<SearchContext.Provider value={userContext}>
				{props.children}
		</SearchContext.Provider>
	)
}

export default SearchContextProvider;
