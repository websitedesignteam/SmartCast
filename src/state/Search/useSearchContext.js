import {useContext} from 'react';
import {SearchContext} from "./SearchContextProvider"
export const useSearchContext = () => {
       const context =useContext(SearchContext);
       if(context === undefined){
              throw new Error("Could not find search context")
       }
       return context;
}
