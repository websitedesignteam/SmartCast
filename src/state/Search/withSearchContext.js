import SearchContextProvider from "./SearchContextProvider"
import React from "react"
export const withSearchContext = (Component) => (props) => (
       <SearchContextProvider>
              <Component {...props}/>
       </SearchContextProvider>
)