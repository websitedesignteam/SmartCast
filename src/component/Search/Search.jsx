import React from 'react'
import {useState} from 'react'
import styles from '../Search/Search.module.scss'
import { useHistory } from 'react-router-dom'
import {useSearchContext} from 'state/Search/useSearchContext'
import { baseUrl, searchTypes } from "../../utils/constants"

function Search(props) {
	//vars
	const searchContext = useSearchContext();
	const history = useHistory();

	//states
	const [query, setQuery] = useState('');
	
	//util functions
	const handleSearch = (event) => {
		setQuery(event.target.value);
	}

	const handleSubmit = (event) =>{
		event.preventDefault();
		if(!searchTypes.includes(searchContext.searchType)){
			alert('Please change your search type');
		} else if (!!query) {
			searchContext.setSearchInput(query);
			history.push(`/search/results/${query}/${searchContext.searchType}`);
		} else {
			alert('Please enter a search term');
		}
	}

	const handleCheckboxes = (value) => {
		if (searchContext.searchType !== value) {
			searchContext.setSearchType(value);
		} else {
			searchContext.setSearchType("tags");
		}
	}

	return (
		<div id="search" className={styles.overlayContent}>
			<form id="search-form" onSubmit={handleSubmit}>
				<div className={styles.searchBar}>
					<img 
						className={styles.searchIcon} 
						src={baseUrl + "/assets/search.svg"} 
						alt="" 
					/>
					<input 
						className={styles.searchInput} 
						type="text" 
						placeholder="Search for Podcasts" 
						name="search" 
						onChange={handleSearch}
					/>
				</div>
				<div className={styles.checkboxes}>
					<label>
						<input 
							type="checkbox" 
							name="episodes" 
							onChange={()=>handleCheckboxes('episodes')} 
							checked={searchContext.searchType==="episodes"} 
						/>
						Include all episodes
					</label>
					<label>
						<input 
							type="checkbox" 
							name="podcasts" 
							onChange={()=>handleCheckboxes('podcasts')} 
							checked={searchContext.searchType==="podcasts"} 
						/>
						Include all podcasts
					</label>
				</div>
				<input type="submit" className={styles.searchEnter}/>
			</form>
		</div>
			
	)
}

export default Search;
