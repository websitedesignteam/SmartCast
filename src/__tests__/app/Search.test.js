import React from 'react';
import { shallow } from 'enzyme';
import Search from 'src/component/Search/Search';
import SearchContextProvider from 'src/state/Search/SearchContextProvider'

test("should render the Search", () => {
    const props = {
        searchContext: {
            searchInput: "",
            searchType: "",
        }
    }

    const searchComponent = shallow(
        <SearchContextProvider {...props}>
            <Search />
        </SearchContextProvider>)
    expect(searchComponent.getElements()).toMatchSnapshot();
});