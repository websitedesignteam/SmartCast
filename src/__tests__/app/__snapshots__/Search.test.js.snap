// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Search 1`] = `
Array [
  <Context.Provider
    value={
      Object {
        "disableCheckbox": true,
        "searchInput": "",
        "searchType": "tags",
        "setDisableCheckbox": [Function],
        "setSearchInput": [Function],
        "setSearchType": [Function],
      }
    }
  >
    <Search />
  </Context.Provider>,
]
`;
